File: android/proj.android/src/cn/htgames/doudizhu/GameInfo.java
package cn.htgames.doudizhu;

import org.json.JSONException;
import org.json.JSONObject;

/**
 * Created by rockyl on 16/9/11.
 */
public class GameInfo {
	public String uid;
	public String serverType;
	public String token;

	public void update(JSONObject data){
		try {
			uid = data.getString("uid");
			serverType = data.getString("serverType");
			token = data.getString("token");
		} catch (JSONException e) {
			e.printStackTrace();
		}
	}

	private static GameInfo _instance;
	public static GameInfo getInstance(){
		if(_instance == null){
			_instance = new GameInfo();
		}
		return _instance;
	}
}

File: ios/newplatfrom/GameInfo.m
//
//  GameInfo.m
//  doudizhuhero
//
//  Created by 劳琪峰 on 16/7/6.
//  Copyright © 2016年 htgames. All rights reserved.
//

#import "GameInfo.h"

static GameInfo *instance;
@implementation GameInfo
+(GameInfo *)getInstance{
	if(instance == nil){
		instance = [GameInfo alloc];
	}
	return instance;
}

-(void)update:(NSDictionary *)data{
	id value = data[@"uid"];
	if([value isKindOfClass:[NSNumber class]]){
		self.uid = [value stringValue];
	}else{
		self.uid = value;
	}
    self.serverType = data[@"serverType"];
    self.token = data[@"token"];
}
@end
